### PR TITLE
Section 3.5 rewrite

### PR DESCRIPTION
**Pull request by gselander**

### CODE REVIEW

**Comment by gselander** [comment]
John requested the basic authentication services by EDHOC being highlighted in the beginning of the document. I tried to fit that in but didn't find a good place. Here is some text if someone wants to try:


> EDHOC performs proof-of-possession that the other endpoint has the private key corresponding to its public key. EDHOC also transports information about the authentication credential (e.g., X.509 certificate or CBOR Web Token) containing the public key. Other authentication related verifications are out of scope for EDHOC, and is the responsibility of the application.

**Comment by emanjon** [comment]
A lot of changes are good and needed. Comments

Hard to get an overview of this PR. Especially which of the deleted text that has been removed and what has just been moved around.

>EDHOC MUST allow the application to read the authentication credential.
Should this be "information about authentication credentials", i.e. ID_CRED_x?

>EDHOC MUST have access to the authentication key of the authentication credential.
EDHOC also need the authentication credentials, i.e. CRED_x?

- Not sure all the EAD text belongs in the body.
  - Lot of duplication in processing.
  - A lot of the text is talking about unfinished drafts.

- The new text is Authentication Related Verifications {#auth-validation} has several issues.

- "match the algorithm"
- "The validation needs to be made"
- "The issuer of the authentication credential needs to be trusted with issuing the credential."
- "the trust anchor is a Certification Authority (CA) certificate"
- "at least one CA public key certificate"
- "subject of the certificate owns the public key"

I am also not sure it should be added. The comment was to shorten down 3.5. This seems to expand things.

**Review by emanjon — APPROVED** [review]

File: draft-ietf-lake-edhoc.md [hunk 1]
```diff
@@ -389,54 +392,42 @@ Connection identifiers may be used to correlate EDHOC messages and facilitate th
...
+    * The authentication key of the authentication credential is used with the Signature_or_MAC field to verify proof-of-possession of the private key.
 
-The authentication credential (CRED_x) contains, in addition to the authentication key, also the authentication key algorithm and optionally other parameters such as identity, key usage, expiry, issuer, etc. (see {{auth-cred}}). Identical authentication credentials need to be established in both endpoints to be able to verify integrity, see {{fig-sigma}}. For many settings it is not necessary to transport the authentication credential within EDHOC over constrained links, for example, it may be pre-provisioned or acquired out-of-band over less constrained links.
+Other authentication related verifications are out of scope for EDHOC, and is the responsibility of the application. In particular, the authentication credential needs to be validated in the context of the connection for which EDHOC is used, see {{auth-validation}}. EDHOC MUST allow the application to read received information about credential (ID_CRED_R, ID_CRED_I). EDHOC MUST have access to the authentication key and the authentication credential.
```
Comment (malishav) [inline draft-ietf-lake-edhoc.md:404]: Appendix D contains the information I was missing when going through this paragraph. However, the sentence referencing Appendix D is a bit obscure and did not give me an incentive to go read it. Can we rephrase as follows:

s/see {{auth-validation}}/.For examples, see {{auth-validation}.

File: draft-ietf-lake-edhoc.md [hunk 2]
```diff
@@ -1805,8 +1804,91 @@ When ID_CRED_x does not contain the actual credential, it may be very short, e.g
 
 Note that a COSE header map can contain several header parameters, for example { x5u, x5t } or { kid, kid_context }.
 
-ID_CRED_x MAY also identify the authentication credential by value. For example, a certificate chain can be transported in ID_CRED_x with COSE header parameter c5c or x5chain, defined in {{I-D.ietf-cose-cbor-encoded-cert}} and {{I-D.ietf-cose-x509}} and credentials of type CWT and CCS can be transported with the COSE header parameters registered in {{cwt-header-param}}.
+ID_CRED_x MAY also identify the credential by value. For example, a certificate chain can be transported in ID_CRED_x with COSE header parameter c5c or x5chain, defined in {{I-D.ietf-cose-cbor-encoded-cert}} and {{I-D.ietf-cose-x509}} and credentials of type CWT and CCS can be transported with the COSE header parameters registered in {{cwt-header-param}}.
+
+
+# Authentication Related Verifications {#auth-validation}
```
Comment (malishav) [inline draft-ietf-lake-edhoc.md:1810]: A proposal to rename the title of the appendix to:

# Out-of-scope Authentication-related Verifications {#auth-validation}

File: draft-ietf-lake-edhoc.md [hunk 3]
```diff
@@ -389,54 +392,42 @@ Connection identifiers may be used to correlate EDHOC messages and facilitate th
...
+Other authentication related verifications are out of scope for EDHOC, and is the responsibility of the application. In particular, the authentication credential needs to be validated in the context of the connection for which EDHOC is used, see {{auth-validation}}. EDHOC MUST allow the application to read received information about credential (ID_CRED_R, ID_CRED_I). EDHOC MUST have access to the authentication key and the authentication credential.
 
-EDHOC relies on COSE for identification of authentication credentials (using ID_CRED_x, see {{id_cred}}) and supports all credential types for which COSE header parameters are defined (see {{auth-cred}}).
+Note that the type of authentication key, authentication credential, and the identification of the credential have a large impact on the message size. For example, the signature_or_MAC field is much smaller with a static DH key than with a signature key. A CCS is much smaller than a self-signed certificate/CWT, but if it is possible to reference the credential with a COSE header like 'kid', then that is in turn much smaller than a CCS.
```
Comment (malishav) [inline draft-ietf-lake-edhoc.md:406]: s/signature_or_MAC/Signature_or_MAC

File: draft-ietf-lake-edhoc.md [hunk 4]
```diff
@@ -389,54 +392,42 @@ Connection identifiers may be used to correlate EDHOC messages and facilitate th
...
-
-* When PKI is not used (CCS, self-signed certificate/CWT), the trust anchor is the authentication key of the other party. In this case, the identity is typically directly associated to the authentication key of the other party. For example, the name of the subject may be a canonical representation of the public key. Alternatively, if identities can be expressed in the form of unique subject names assigned to public keys, then a binding to identity can be achieved by including both public key and associated subject name in the protocol message computation: CRED_I or CRED_R may be a self-signed certificate/CWT or CCS containing the authentication key and the subject name, see {{auth-cred}}. In order to run EDHOC, each endpoint needs a specific authentication key/unique associated subject name, or a set of public authentication keys/unique associated subject names, which it is allowed to communicate with. EDHOC provides the proof that the other party possesses the private authentication key corresponding to the public authentication key.
+### Authentication Keys {#auth-keys}
```
Comment (malishav) [inline draft-ietf-lake-edhoc.md:408]: I don't understand why this section is presented independently from the presentation of the EDHOC method in Section 3.2

**Review by malishav — COMMENTED** [review]
Thanks for doing these changes. Section 3.5 reads nice. A couple of comments inline.

**Pull request merged**